From 1897d860b696f0f016fcfe9d52db77ab0d72c5a8 Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Fri, 9 Nov 2018 17:11:53 -0800
Subject: [PATCH 6/6] lineage-sdk: Move styles API from public to internal

*) Move StyleInterface from public to internal

*) Remove styles from public API

*) Replace lineageos.permission.CHANGE_STYLE with
   AOSP android.permission.CHANGE_OVERLAY_PACKAGES.

TODO: remove permlab_changeStyle (and permdesc) strings
      what else ?

Change-Id: I2dfb339688795ad5a7ae70d8d60a7dc51bbdb4cb
---
 api/lineage_current.txt                       | 34 -------------------
 .../internal/StyleInterfaceService.java       |  6 ++--
 lineage/res/AndroidManifest.xml               |  7 ----
 .../internal}/style/IStyleInterface.aidl      |  6 ++--
 .../internal}/style/StyleInterface.java       |  4 +--
 .../lineageos/internal}/style/Suggestion.aidl |  2 +-
 .../lineageos/internal}/style/Suggestion.java |  2 +-
 7 files changed, 10 insertions(+), 51 deletions(-)
 rename sdk/src/java/{lineageos => org/lineageos/internal}/style/IStyleInterface.aidl (92%)
 rename sdk/src/java/{lineageos => org/lineageos/internal}/style/StyleInterface.java (99%)
 rename sdk/src/java/{lineageos => org/lineageos/internal}/style/Suggestion.aidl (94%)
 rename sdk/src/java/{lineageos => org/lineageos/internal}/style/Suggestion.java (98%)

diff --git a/api/lineage_current.txt b/api/lineage_current.txt
index cb4b518..bd93fc1 100644
--- a/api/lineage_current.txt
+++ b/api/lineage_current.txt
@@ -362,7 +362,6 @@ package lineageos.platform {
     ctor public Manifest.permission();
     field public static final java.lang.String ACCESS_WEATHER_MANAGER = "lineageos.permission.ACCESS_WEATHER_MANAGER";
     field public static final java.lang.String BIND_WEATHER_PROVIDER_SERVICE = "lineageos.permission.BIND_WEATHER_PROVIDER_SERVICE";
-    field public static final java.lang.String CHANGE_STYLE = "lineageos.permission.CHANGE_STYLE";
     field public static final java.lang.String HARDWARE_ABSTRACTION_ACCESS = "lineageos.permission.HARDWARE_ABSTRACTION_ACCESS";
     field public static final java.lang.String MANAGE_LIVEDISPLAY = "lineageos.permission.MANAGE_LIVEDISPLAY";
     field public static final java.lang.String MANAGE_REMOTE_PREFERENCES = "lineageos.permission.MANAGE_REMOTE_PREFERENCES";
@@ -815,10 +814,6 @@ package lineageos.providers {
     field public static final java.lang.String BATTERY_LIGHT_LOW_COLOR = "battery_light_low_color";
     field public static final java.lang.String BATTERY_LIGHT_MEDIUM_COLOR = "battery_light_medium_color";
     field public static final java.lang.String BATTERY_LIGHT_PULSE = "battery_light_pulse";
-    field public static final java.lang.String BERRY_CURRENT_ACCENT = "berry_current_accent";
-    field public static final java.lang.String BERRY_DARK_OVERLAY = "berry_dark_overlay";
-    field public static final java.lang.String BERRY_GLOBAL_STYLE = "berry_global_style";
-    field public static final java.lang.String BERRY_MANAGED_BY_APP = "berry_managed_by_app";
     field public static final java.lang.String BLUETOOTH_ACCEPT_ALL_FILES = "bluetooth_accept_all_files";
     field public static final java.lang.String BUTTON_BACKLIGHT_ONLY_WHEN_PRESSED = "button_backlight_only_when_pressed";
     field public static final java.lang.String CALL_RECORDING_FORMAT = "call_recording_format";
@@ -1003,35 +998,6 @@ package lineageos.providers {
 
 }
 
-package lineageos.style {
-
-  public class StyleInterface {
-    method public java.lang.String getAccent();
-    method public int getGlobalStyle();
-    method public static lineageos.style.StyleInterface getInstance(android.content.Context);
-    method public lineageos.style.Suggestion getSuggestion(android.graphics.Bitmap, int[]);
-    method public java.util.List<java.lang.String> getTrustedAccents();
-    method public boolean isDarkNow();
-    method public boolean setAccent(java.lang.String);
-    method public boolean setGlobalStyle(int, java.lang.String);
-    field public static final java.lang.String ACCENT_DEFAULT = "lineageos";
-    field public static final java.lang.String CHANGE_STYLE_SETTINGS_PERMISSION = "lineageos.permission.CHANGE_STYLE";
-    field public static final int STYLE_GLOBAL_AUTO_DAYTIME = 1; // 0x1
-    field public static final int STYLE_GLOBAL_AUTO_WALLPAPER = 0; // 0x0
-    field public static final int STYLE_GLOBAL_DARK = 3; // 0x3
-    field public static final int STYLE_GLOBAL_LIGHT = 2; // 0x2
-  }
-
-  public class Suggestion implements android.os.Parcelable {
-    ctor public Suggestion(int, int);
-    method public int describeContents();
-    method public void writeToParcel(android.os.Parcel, int);
-    field public final int globalStyle;
-    field public final int selectedAccent;
-  }
-
-}
-
 package lineageos.trust {
 
   public class TrustInterface {
diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java b/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java
index edc36ba..3166380 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/StyleInterfaceService.java
@@ -36,10 +36,10 @@ import com.android.server.SystemService;
 
 import lineageos.app.LineageContextConstants;
 import lineageos.providers.LineageSettings;
-import lineageos.style.IStyleInterface;
-import lineageos.style.StyleInterface;
-import lineageos.style.Suggestion;
 import lineageos.util.palette.Palette;
+import org.lineageos.internal.style.IStyleInterface;
+import org.lineageos.internal.style.StyleInterface;
+import org.lineageos.internal.style.Suggestion;
 
 import java.util.ArrayList;
 import java.util.List;
diff --git a/lineage/res/AndroidManifest.xml b/lineage/res/AndroidManifest.xml
index aa4e0f5..ed70182 100644
--- a/lineage/res/AndroidManifest.xml
+++ b/lineage/res/AndroidManifest.xml
@@ -117,13 +117,6 @@
                 android:icon="@drawable/ic_launcher_lineageos"
                 android:protectionLevel="normal" />
 
-    <!-- Allows an application to change system style -->
-    <permission android:name="lineageos.permission.CHANGE_STYLE"
-                android:label="@string/permlab_changeStyle"
-                android:description="@string/permdesc_changeStyle"
-                android:icon="@drawable/ic_launcher_lineageos"
-                android:protectionLevel="dangerous" />
-
     <!-- Allows an application to access the Trust interface -->
     <permission android:name="lineageos.permission.TRUST_INTERFACE"
                 android:label="@string/permlab_trustInterface"
diff --git a/sdk/src/java/lineageos/style/IStyleInterface.aidl b/sdk/src/java/org/lineageos/internal/style/IStyleInterface.aidl
similarity index 92%
rename from sdk/src/java/lineageos/style/IStyleInterface.aidl
rename to sdk/src/java/org/lineageos/internal/style/IStyleInterface.aidl
index 514766e..57d9638 100644
--- a/sdk/src/java/lineageos/style/IStyleInterface.aidl
+++ b/sdk/src/java/org/lineageos/internal/style/IStyleInterface.aidl
@@ -15,10 +15,10 @@
 ** limitations under the License.
 */
 
-package lineageos.style;
+package org.lineageos.internal.style;
 
 import android.graphics.Bitmap;
-import lineageos.style.Suggestion;
+import org.lineageos.internal.style.Suggestion;
 
 import java.util.List;
 
@@ -33,4 +33,4 @@ interface IStyleInterface {
     boolean isDarkNow();
     boolean setDarkOverlay(String overlayName);
     String getDarkOverlay();
-}
\ No newline at end of file
+}
diff --git a/sdk/src/java/lineageos/style/StyleInterface.java b/sdk/src/java/org/lineageos/internal/style/StyleInterface.java
similarity index 99%
rename from sdk/src/java/lineageos/style/StyleInterface.java
rename to sdk/src/java/org/lineageos/internal/style/StyleInterface.java
index 9ff9255..91eb3c7 100644
--- a/sdk/src/java/lineageos/style/StyleInterface.java
+++ b/sdk/src/java/org/lineageos/internal/style/StyleInterface.java
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package lineageos.style;
+package org.lineageos.internal.style;
 
 import android.content.Context;
 import android.graphics.Bitmap;
@@ -93,7 +93,7 @@ public class StyleInterface {
      * it at runtime as any other dangerous permission
      */
     public static final String CHANGE_STYLE_SETTINGS_PERMISSION =
-            "lineageos.permission.CHANGE_STYLE";
+            "android.permission.CHANGE_OVERLAY_PACKAGES";
 
     private static final String TAG = "StyleInterface";
 
diff --git a/sdk/src/java/lineageos/style/Suggestion.aidl b/sdk/src/java/org/lineageos/internal/style/Suggestion.aidl
similarity index 94%
rename from sdk/src/java/lineageos/style/Suggestion.aidl
rename to sdk/src/java/org/lineageos/internal/style/Suggestion.aidl
index 242e6d1..6b03fbd 100644
--- a/sdk/src/java/lineageos/style/Suggestion.aidl
+++ b/sdk/src/java/org/lineageos/internal/style/Suggestion.aidl
@@ -15,6 +15,6 @@
 ** limitations under the License.
 */
 
-package lineageos.style;
+package org.lineageos.internal.style;
 
 parcelable Suggestion;
diff --git a/sdk/src/java/lineageos/style/Suggestion.java b/sdk/src/java/org/lineageos/internal/style/Suggestion.java
similarity index 98%
rename from sdk/src/java/lineageos/style/Suggestion.java
rename to sdk/src/java/org/lineageos/internal/style/Suggestion.java
index 314797e..965115a 100644
--- a/sdk/src/java/lineageos/style/Suggestion.java
+++ b/sdk/src/java/org/lineageos/internal/style/Suggestion.java
@@ -15,7 +15,7 @@
 ** limitations under the License.
 */
 
-package lineageos.style;
+package org.lineageos.internal.style;
 
 import android.os.Parcel;
 import android.os.Parcelable;
-- 
2.17.1

